models/base.py

```diff
@@ -166,6 +166,10 @@ def __call__(self, *args, name: Optional[str] = None, **kwargs) -> Layer:
       if self.calls:
         name_ctx.is_repeated_call = True
       layer_dict = self.make_layer_dict(*args, **kwargs)
+      ret = None
+      if isinstance(layer_dict, Tuple):
+        ret = layer_dict[1:]
+        layer_dict = layer_dict[0]
```

Comment:
```suggestion
      if not isinstance(layer_dict, dict):
        ret, layer_dict = layer_dict
        assert isinstance(layer_dict, dict)
```

Reply:
I am not sure if `layer_dict, ret = layer_dict` works in the general case, where possibly there is more than just one additional layer passed outside of the subnetwork, shouldn't this run into an error in a case where `len(layer_dict)==3` ?

Reply:
`ret` itself would be the tuple, or whatever other structure.

Reply:
Ah I see. 👍